server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/UseCaseEngine.java

```diff
@@ -447,6 +458,33 @@ private Map<String, List<Location>> extractDefinitions(AnalysisResult result, No
                 || ImplicitCodeUtils.isImplicit(context.getDefinitions().get(0).getUri())));
   }
 
+  private Map<String, List<Location>> extractFunctionDefinitions(AnalysisResult result) {
+   return  result
+            .getRootNode()
+            .getDepthFirstStream()
+            .filter(hasType(REFERENCE).or(hasType(FUNCTION_REFERENCE)))
+            .map(UsageNode.class::cast)
+            .filter(u -> u.getUsageType() == UsageNode.UsageType.FUNCTION)
+            .map(UsageNode::getDefinitionNodes)
+            .flatMap(Collection::stream)
+            .map(DefinedAndUsedStructure.class::cast)
+            .filter(context -> !(context.getDefinitions().isEmpty()
+                    || ImplicitCodeUtils.isImplicit(context.getDefinitions().get(0).getUri())))
+            .collect(toMap(DefinedAndUsedStructure::getDefinitions, REFERENCE));
+  }
+
+  private Map<String, List<Location>> extractDefinitionsUsage(AnalysisResult result) {
+    return  result
+            .getRootNode()
+            .getDepthFirstStream()
+            .filter(hasType(REFERENCE).or(hasType(FUNCTION_REFERENCE)))
+            .map(UsageNode.class::cast)
+            .filter(u -> u.getUsageType() == UsageNode.UsageType.FUNCTION)
+            .map(DefinedAndUsedStructure.class::cast)
+            .filter(context -> !context.getDefinitions().isEmpty())
+            .collect(toMap(d -> ImmutableList.of(d.getLocality().toLocation()), REFERENCE));
```

Comment:
Shouldn't this at some point call `getUsages` to verify that the usages are collected correctly?